src/server/pkg/obj/cache_client.go

```diff
@@ -40,28 +40,52 @@ func NewCacheClient(slow, fast Client, size int) Client {
 }
 
 func (c *cacheClient) Reader(ctx context.Context, p string, offset, size uint64) (io.ReadCloser, error) {
-	c.doPopulateOnce(ctx)
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	if _, exists := c.cache.Get(p); exists {
-		return c.fast.Reader(ctx, p, offset, size)
+	cacheP := cachePath(p, offset, size)
```

Comment:
@Tryneus A change I needed to make to our cache client for 2.0 was to make the support for offset / size actually cache just the offset / size rather than the full object. This is unnecessary for 2.0 since we only operate on chunks with a pre-defined max size, but for 1.x the object can be the full hashtree which may be too big to cache in the local filesystem (particularly since we can have multiple hashtrees in the cache). The change I made is intended to store the path, offset, size as the key in the local cache so we only download and cache the object parts that are relevant to the request (this should work well with the path range indexing that is used for output repos).